pod-operation/src/components/pressure_transducer.rs

```diff
@@ -2,51 +2,101 @@ use ina219::INA219;
 use rppal::i2c::I2c;
 use tracing::debug;
 
-pub struct PressureTransducer {
-	ina: INA219<I2c>,
-}
-
 // The calibration value is used to adjust the maximum current measurement
 // and precision of measurements.
 const INA219_CALIBRATION_VALUE: u16 = 0xffff;
 
+// The pod will be using two INA219s, so we'll need to differentiate them
+// via their device address.
+const INA219_UPSTREAM_ADDRESS: u8 = 0x40;
+const INA219_DOWNSTREAM_ADDRESS: u8 = 0x41;
+
 // Even with the calibration values, the readings from the INA219 are not in
 // mA. A scaling factor is needed in order to convert the raw reading to mA and
 // this is not provided in the INA219 library that we are using. Note that this
 // value changes according to the calibration value. The exact formula can be
 // found in the INA219 datasheet.
 const INA219_SCALING_VALUE: f32 = 160.0;
 
-// The pressure transducer outputs a current between 4 mA and 20 mA with 0 PSI
-// and 300 PSI respectively. Assuming a linear interpolation, a 1 mA increase
-// results in a 18.75 PSI increase.
-const REF_CURRENT_LOW: f32 = 4.0;
-const REF_CURRENT_HIGH: f32 = 20.0;
-const REF_PRESSURE_LOW: f32 = 0.0;
-const REF_PRESSURE_HIGH: f32 = 300.0;
+struct Reference {
+	pressure_lo: f32,
+	pressure_hi: f32,
+	current_lo: f32,
+	current_hi: f32,
+}
 
-const REF_CURRENT_SPAN: f32 = REF_CURRENT_HIGH - REF_CURRENT_LOW;
-const REF_PRESSURE_SPAN: f32 = REF_PRESSURE_HIGH - REF_PRESSURE_LOW;
+// The upstream pressure transducer outputs a current between 4 mA and 20 mA
+// with 0 PSI and 300 PSI respectively.
+const UPSTREAM_REF: Reference = Reference {
+	pressure_lo: 0.0,
+	pressure_hi: 300.0,
+	current_lo: 4.0,
+	current_hi: 20.0,
+};
 
-impl PressureTransducer {
-	pub fn new(ina219_addr: u8) -> Self {
+// The downtream pressure transducer outputs a current between 4 mA and 20 mA
+// with 0 PSI and 300 PSI respectively.
+const DOWNSTREAM_REF: Reference = Reference {
+	pressure_lo: 0.0,
+	pressure_hi: 300.0,
+	current_lo: 4.0,
+	current_hi: 20.0,
+};
+
+struct CalibratedINA;
+
+impl CalibratedINA {
+	fn init_ina(device_address: u8) -> INA219<I2c> {
 		let device = I2c::new().unwrap();
 
-		let mut ina219 = INA219::new(device, ina219_addr);
+		let mut ina219 = INA219::new(device, device_address);
 		debug!("Initialized I2C and INA219");
 
 		ina219.calibrate(INA219_CALIBRATION_VALUE).unwrap();
 		debug!("Calibrating INA219");
 
-		PressureTransducer { ina: ina219 }
+		ina219
+	}
+}
+
+pub struct PressureTransducer {
+	ina: INA219<I2c>,
+	ref_measurements: Reference,
```

Comment:
Nitpick: unsure if measurements is an appropriate term since these reference values are more so characteristic to the transducer, not necessarily measured. Perhaps consider `ref_values`?